tools/benchmark.py

```diff
@@ -22,7 +22,7 @@
 exec_time_benchmarks = [
     ("hello", ["run", "cli/tests/002_hello.ts"]),
     ("relative_import", ["run", "cli/tests/003_relative_import.ts"]),
-    ("error_001", ["run", "cli/tests/error_001.ts"]),
+    ("error_001", ["run", "cli/tests/error_001.ts", "|| true"]),
```

Comment:
The `"|| true"` is kind of hacky. It's not an argument. It seems to work, but I fear this could cause some frustration in the future. I think it would be better to change the tuple to contain a boolean indicating whether each should fail or pass.

Maybe something like this?
```py
exec_time_benchmarks = [
    ("hello", True, ["run", "cli/tests/002_hello.ts"]),
    ("relative_import", True, ["run", "cli/tests/003_relative_import.ts"]),
    ("error_001", False, ["run", "cli/tests/error_001.ts"]),
```

Reply:
I'm particularly concerned about bits like this:
```py
    for (name, args) in exec_time_benchmarks:
        s = get_strace_summary([deno_exe] + args)
```
The `|| true` is only working coincidentally.

Reply:
I agree with you. Adding `|| true` to the arguments isn't exactly what we need.  

I updated the PR and to include an optional expected return code to the benchmark scenarios.  You can specify the expected Error code or `None` to retain the default behavior.  

`hyperfine` takes a list of commands to benchmark.  When we remove the `--ignore-failure` flag, it will exit as soon as it encounters an error so we'll need some mechanism to capture at failure.  Appending `|| true` would give just swallow any errors.  I added `; test $? -eq 1` which tests that the last return code (`$?`) is equal to the expected return/exit code.

Reply:
@ry I updated the PR with your feedback.  Let me know what you think